pkg/inject/envoy.go

```diff
@@ -166,3 +170,23 @@ func (m *envoyMutator) getPreview(pod *corev1.Pod) string {
 	}
 	return "0"
 }
+
+// containsEnvoyContainer checks whether pod already contains "envoy" container
+func containsEnvoyContainer(pod *corev1.Pod) bool {
+	for _, container := range pod.Spec.Containers {
+		if container.Name == envoyContainerName {
```

Comment:
Since this a name check, would this have an impact on envoy version bump workflows?

Reply:
it don't. since pod spec can only be modified at pod creation time. :D